### DISCUSSION TITLE
ECM/RNDIS support/example? (network via USB)

### DISCUSSION THREAD

**Discussion by eadmaster** [espressif/arduino-esp32#10157] [Q&A]
is it possible to set a virtual [Ethernet](https://en.wikipedia.org/wiki/Ethernet) link on boards like the ESP32S3?

i've found some references here, but none in this repo:

 - https://github.com/espressif/esp-iot-solution/blob/master/examples/usb/device/usb_dongle/README_EN.md
 - https://github.com/hathach/tinyusb/discussions/742

**Comment by me-no-dev** [comment]
we will add support soon. Not yet available

**Reply by 7h30th3r0n3** [reply]
any news about this ?

**Reply by xxxxxixxxxx** [reply]
> we will add support soon. Not yet available

any news ?

**Comment by 0x0fe** [comment]
Hello, is there any update regarding USB RNDIS support in the core? it's a components of the IDF, it would be nice if we could either have it in the core, or as a separate lib
